### PR TITLE
Add RateLimiter APIs to the server

### PR DESCRIPTION
**Pull request by tanmayv25**
See this PR for previous reviews: https://github.com/triton-inference-server/core/pull/6

Linked Server PR: https://github.com/triton-inference-server/server/pull/2878/files

### CODE REVIEW

File: include/triton/core/tritonserver.h [hunk 1]
```diff
@@ -1233,6 +1239,42 @@ TRITONSERVER_DECLSPEC TRITONSERVER_Error*
...
+    TRITONSERVER_ServerOptions* options, TRITONSERVER_RateLimitMode mode);
+
+/// Add resource count for rate limiting.
+///
+/// \param options The server options object.
+/// \param name The name of the resource.
+/// \param count The count of the resource.
+/// \param device The device identifier for the resource. A value of -1
+/// indicates that the specified number of resources are available on every
+/// device. The specification is ignored for a global resource. The server
```
Comment (deadeyegoodwin) [inline include/triton/core/tritonserver.h:1267]: How come ignored for global resource? Can't I set the count for a global resource?

Comment (tanmayv25) [reply include/triton/core/tritonserver.h:1267]: This comment is related to the device parameter. The name and count can be set for all resource types.

Comment (deadeyegoodwin) [reply include/triton/core/tritonserver.h:1267]: "... the device value is ignored for a global resource."

Comment (tanmayv25) [reply include/triton/core/tritonserver.h]: I see how it can be confusing. Fixed the language.

**Review by deadeyegoodwin — APPROVED** [review]

**Pull request merged**